hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/rate/maxrate/MaxRateProviderFactory.java

```diff
@@ -2,34 +2,27 @@
 
 import com.google.common.base.Preconditions;
 import pl.allegro.tech.hermes.api.Subscription;
-import pl.allegro.tech.hermes.common.config.ConfigFactory;
 import pl.allegro.tech.hermes.common.metric.HermesMetrics;
 import pl.allegro.tech.hermes.consumers.consumer.rate.SendCounters;
 
-import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_MAXRATE_BUSY_TOLERANCE;
-import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_MAXRATE_HISTORY_SIZE;
-import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_MAXRATE_MIN_MAX_RATE;
-import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_MAXRATE_MIN_SIGNIFICANT_UPDATE_PERCENT;
-import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_WORKLOAD_NODE_ID;
-
 public class MaxRateProviderFactory {
 
     private final Creator providerCreator;
 
-    public MaxRateProviderFactory(ConfigFactory configFactory,
+    public MaxRateProviderFactory(MaxRateParameters maxRateParameters,
+                                  String nodeId,
                                   MaxRateRegistry maxRateRegistry,
                                   MaxRateSupervisor maxRateSupervisor,
                                   HermesMetrics metrics) {
 
-        checkNegotiatedSettings(configFactory);
+        checkNegotiatedSettings(maxRateParameters.getMinSignificantUpdatePercent() / 100, maxRateParameters.getBusyTolerance());
```

Comment:
what do you think about creating new local variable like it was in `checkNegotiatedSettings` method:
`double checkNegotiatedSettings = maxRateParameters.getMinSignificantUpdatePercent() / 100` and then use it in line 18,23 ?